src/Sentry.OpenTelemetry/SentrySpanProcessor.cs

```diff
@@ -181,6 +181,14 @@ public override void OnEnd(Activity data)
             return;
         }
 
+        // Skip any activities that are not recorded.
+        if (data is { Recorded: false })
+        {
+            _options?.DiagnosticLogger?.LogDebug($"Ignoring unrecorded Activity {data.SpanId}.");
```

Comment:
Should pass in data.SpanId as a separate parameter. This avoids a string.Format and an allocation in scenarios where debugging is disabled.